core/client/fs/src/test/java/alluxio/client/file/FileInStreamTest.java

```diff
@@ -682,63 +641,13 @@ private void testReadBuffer(int dataRead) throws Exception {
     mTestStream.close();
 
     assertArrayEquals(BufferUtils.getIncreasingByteArray(dataRead), buffer);
-    verifyCacheStreams(dataRead);
   }
 
   /**
-   * Verifies that data was properly written to the cache streams.
-   *
-   * @param dataRead the bytes to read
-   */
-  private void verifyCacheStreams(long dataRead) {
-    int cachedData = (int) ((dataRead / BLOCK_LENGTH) * BLOCK_LENGTH);
-    for (int streamIndex = 0; streamIndex < NUM_STREAMS; streamIndex++) {
-      TestBlockInStream inStream = mInStreams.get(streamIndex);
-      TestBlockOutStream stream = mCacheStreams.get(streamIndex);
-      byte[] data = stream.getWrittenData();
-      if (streamIndex * BLOCK_LENGTH > dataRead) {
-        assertEquals(0, data.length);
-      } else {
-        long dataStart = streamIndex * BLOCK_LENGTH;
-        assertEquals(Math.min(BLOCK_LENGTH, dataRead - dataStart), inStream.getBytesRead());
-        if (mBlockSource != BlockInStreamSource.REMOTE) {
-          // no data is written cache stream if the read is not from remote worker
-          assertEquals(0, data.length);
-        } else {
-          for (int i = 0; i < BLOCK_LENGTH && dataStart + i < cachedData; i++) {
-            assertEquals((byte) (dataStart + i), data[i]);
-          }
-        }
-      }
-    }
-  }
-
-  /**
-   * Validates the partial caching behavior given the different block source type. This function
-   * assumes the block at the given index is read and cached for the given sizes.
-   *
+   * Validates the partial caching behavior. This function
+   * verifies the block at the given index is read for the given sizes.
    */
-  private void validatePartialCaching(int index, int readSize, int cacheSize) {
-    switch (mBlockSource) {
-      case LOCAL:
-        // nothing is cached, and the entire block is not read
-        assertEquals(readSize, mInStreams.get(index).getBytesRead());
-        assertArrayEquals(new byte[0], mCacheStreams.get(index).getWrittenData());
-        break;
-      case REMOTE:
-        // the block is cached
-        assertEquals(cacheSize, mInStreams.get(index).getBytesRead());
-        assertArrayEquals(
-            BufferUtils.getIncreasingByteArray(index * (int) BLOCK_LENGTH, cacheSize),
-            mCacheStreams.get(index).getWrittenData());
-        break;
-      case UFS:
-        // nothing is cached, but the entire block is read
-        assertArrayEquals(new byte[0], mCacheStreams.get(index).getWrittenData());
-        assertEquals(cacheSize, mInStreams.get(index).getBytesRead());
-        break;
-      default:
-        throw new IllegalStateException("Unrecognized block source type " + mBlockSource);
-    }
+  private void validatePartialCaching(int index, int readSize) {
+    assertEquals(readSize, mInStreams.get(index).getBytesRead());
```

Comment:
Are we still asserting that partial caching is working (ie. the block gets cached) or only that the FileInStream read behavior is correct (only reading requested bytes)? If we can validate caching that would be ideal but since this is a unit test I'm fine with changing this to validating the input stream behavior.

Reply:
I prefer to go with unit-tests without verifying the async-cache for now. The reason is that
it is quite hard to mock netty RPC currently to intercept the Protos sent over netty channel.
The code will be very messy and hard to maintain. Since the async cache logic is also covered in the FileInStream integration test, I am fine with leaving this as is, until we have a better way to mock netty RPCs.

For now, a TODO is left.

Reply:
Yeah I think it is better to keep the async caching logic tested in integration tests if we need to do more in the instream unit test to validate it.